server/src/Middlewares/gatekeeper.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+import jwt, { JwtPayload } from "jsonwebtoken";
+
+
+const ENCRYPTION_KEY  = "app";
+if (!ENCRYPTION_KEY) {
+    console.error("ENCRYPTION_KEY is not defined.");
+}
+const gateKeeper = (roles: string[] = []) => {
```

Comment:
in the gatekeeper we should add checks for each route as well for more security.
so for example if createUser has 2 body variables being sent. we should check that when calling createUser we only have 2 body being sent. else don't let the check pass